### PR TITLE
VIH-8215 Ejud feature flag enabled/disabled

### PR DESCRIPTION
**Pull request by PrasannaDommalapati**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-8215

### Change description ###
when EJUD Feature Flag is OFF
- VHO will not be able to select eJudiciary accounts for judge role
- VHO will not be able to select eJudiciary accounts for panel member/winger role
- VHO will be able to create AAD panel member/winger accounts. 

when EJUD Feature Flag is ON
- VHO will be able to select eJudiciary accounts for judge role
- VHO will be able to select eJudiciary accounts for panel member/winger role
- VHO will not be able to create AAD panel member/winger accounts. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 1]
```diff
@@ -50,18 +52,23 @@ export class AddParticipantComponent extends AddParticipantBaseDirective impleme
...
         protected bookingService: BookingService,
+        protected featureFlagService: FeatureFlagService,
         protected logger: Logger
     ) {
         super(bookingService, router, videoHearingService, logger);
         this.titleList = searchService.TitleList;
+        featureFlagService
+            .getFeatureFlagByName('EJudFeature')
+            .pipe(first())
+            .subscribe(result => (this.judiciaryRoles = result ? Constants.JudiciaryRoles : []));
```
Comment (omio-ma) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:71]: for the if check you want to use triple equals "===" so it should be like this.judiciaryRoles === result

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.spec.ts [hunk 2]
```diff
@@ -132,6 +138,31 @@ describe('SearchEmailComponent', () => {
...
+        expect(component.errorNotFoundJohEmail).toBeFalsy();
+    });
+
+    it('should show showCreateNewUserWarning when EJudFeature flag is OFF and no results found from search service', () => {
+        component.notFoundParticipant = true;
+
+        component.ngOnInit();
+        fixture.detectChanges();
+        component.hearingRoleParticipant = 'Panel Member';
+        expect(component.showCreateNewUserWarning).toBeTruthy();
```
Comment (omio-ma) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.spec.ts:163]: if showCreateNewUserWarning is a boolean we should use the toBe(true) assertion as toBeTruthy will pass if that property is simply defined

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts [hunk 3]
```diff
@@ -48,10 +49,22 @@ export class SearchEmailComponent implements OnInit, OnDestroy {
...
+        private featureFlagService: FeatureFlagService
+    ) {}
 
     ngOnInit() {
         this.email = this.initialValue;
+        this.featureFlagService
+            .getFeatureFlagByName('EJudFeature')
+            .pipe(first())
+            .subscribe(result => {
+                this.judiciaryRoles = result ? Constants.JudiciaryRoles : [];
```
Comment (omio-ma) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts:65]: I am not sure how this has worked? are we setting judiciaryRoles property here or checking the value against result? If its a condition we should use triple equals here

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts (hunk 1 above)
Comment (PrasannaDommalapati) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:71]: here i am not comparing this.judiciary roles and result, i setting value to this.judiciaryRoles

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts (hunk 3 above)
Comment (PrasannaDommalapati) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts:65]: explained in the above comment

**Review by omio-ma — APPROVED** [review]

**Review by ZJianWang — APPROVED** [review]
LGTM

**Pull request merged**